components/03-modules/slider/slider.js

```diff
@@ -6,24 +6,103 @@
   const sliders = [...document.querySelectorAll('.slider')];
 
   sliders.forEach(slider => {
-    const dataValues = slider.dataset;
+    const dataValues   = slider.dataset,
+          items        = slider.querySelectorAll('.slider__item').length,
+          dotContainer = slider.querySelector('.slider__dot-navigation');
 
     let config = {
       classNameFrame: 'slider__handler',
       classNameSlideContainer: 'slider__slides',
       classNamePrevCtrl: 'slider__prev',
-      classNameNextCtrl: 'slider__next'
+      classNameNextCtrl: 'slider__next',
+      slidesToScroll: 1
     };
 
-    for (const key of Object.keys(dataValues)) {
-      if (!isNaN(parseInt(dataValues[key]))) {
-        config[key] = parseInt(dataValues[key]);
+    Object.keys(dataValues).map((objectKey) => {
+      if (objectKey != 'slidesToScroll') {
+        if (!isNaN(parseInt(dataValues[objectKey]))) {
+          config[objectKey] = parseInt(dataValues[objectKey]);
+        }
+        else {
+          config[objectKey] = dataValues[objectKey];
+        }
       }
-      else {
-        config[key] = dataValues[key];
+    });
+
+    function cloneDots() {
+      const dots      = dotContainer.childNodes,
+            dotActive = 'slider__dot--active',
+            dotCount  = dotNumber(),
+            dotItem   = document.createElement('div');
+
+      for (let i = 0; i < dotCount; i++) {
+        let clone = dotItem.cloneNode();
+        const nestedElement = document.createElement('span');
+
+        clone.classList.add('slider__dot');
+        clone.appendChild(nestedElement);
+        nestedElement.classList.add('slider__dot-inside');
+        dotContainer.appendChild(clone);
       }
+
+      dots[0].classList.add(dotActive);
+    }
+
+    function addDotListeners() {
+      const dots = dotContainer.childNodes;
+
+      dots.forEach(dot => {
+        dot.addEventListener('click', (e) => {
+          if (e.target.parentNode === dot) {
+            lorySlider.slideTo(Array.prototype.indexOf.call(dots, e.target.parentNode));
+          }
+          else {
+            lorySlider.slideTo(Array.prototype.indexOf.call(dots, e.target));
+          }
+        });
+      })
+    }
+
+    function dotClassToggle(e) {
```

Comment:
what is `e`? Is it `element` or `event`? Better use `el` id element.

Reply:
renamed to event

Reply:
other `e` sign too